src/api-service/__app__/onefuzzlib/tasks/main.py

```diff
@@ -294,22 +281,40 @@ def on_start(self) -> None:
     def key_fields(cls) -> Tuple[str, str]:
         return ("job_id", "task_id")
 
-    def set_state(self, state: TaskState, send: bool = True) -> None:
-        if self.state == state:
-            return
-
+    def set_state(self, state: TaskState) -> None:
         self.state = state
         if self.state in [TaskState.running, TaskState.setting_up]:
             self.on_start()
 
         self.save()
 
-        send_event(
-            EventTaskStateUpdated(
-                job_id=self.job_id,
-                task_id=self.task_id,
-                state=self.state,
-                end_time=self.end_time,
-                config=self.config,
+        if self.state == TaskState.stopped:
+            if self.error:
+                send_event(
+                    EventTaskFailed(
```

Comment:
we were sending this event when the state was "stopping" instead of "stopped". Is this change intentional ?

Reply:
Yes.  While I was writing this PR, I had shopped a task, got the task stopping event, but files from the task continued to flow in as the agent had stopped the task yet.

Reply:
I pulled the send_state related changes, and I'll make a new PR for that.

Reply:
Note, this additional code is part of #651